src/libraries/LiquidityAmounts.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {FullMath} from "@uniswap/v4-core/src/libraries/FullMath.sol";
+import {FixedPoint96} from "@uniswap/v4-core/src/libraries/FixedPoint96.sol";
+import {SafeCast} from "@uniswap/v4-core/src/libraries/SafeCast.sol";
+
+/// @notice Provides functions for computing liquidity amounts from token amounts and prices
+library LiquidityAmounts {
```

Comment:
diffed with v3 lib, can wrap the functions in unchecked blocks; subtractions guaranteed to not overflow